dali/pipeline/executor/executor.cc

```diff
@@ -644,7 +649,45 @@ bool Executor<WorkspacePolicy, QueuePolicy>::HasConditionals() const {
   return has_conditionals_;
 }
 
+template<typename WorkspacePolicy, typename QueuePolicy>
+void Executor<WorkspacePolicy, QueuePolicy>::InitCheckpointing() {
+  if (!checkpointing_)
+    return;
+
+  // TODO(mstaniewski): verify
+  int queue_depth = stage_queue_depths_[OpType::CPU] + 1;
```

Comment:
I wonder if we should just extnd the `IterationData` struct, it already computes the queue_depth (the same way I think), has the queue, keeps the indicies of producers and consumers.